extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/context/TransactionContext.java

```diff
@@ -0,0 +1,213 @@
+            return instanceHandle.get();
+        } else if (creationalContext != null) {
+            Transaction currentTransaction = getCurrentTransaction();
+            T createdInstance = contextual.create(creationalContext);
+            instanceHandle = new ContextInstanceHandleImpl<>((InjectableBean<T>) contextual, createdInstance,
+                    creationalContext);
+
+            contextState.put(contextual, instanceHandle);
+
+            return createdInstance;
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public <T> T get(Contextual<T> contextual) {
+        return get(contextual, null);
+    }
+
+    /**
+     * The transaction scoped context is active when a transaction is active.
+     */
+    @Override
+    public boolean isActive() {
+        Transaction transaction = getCurrentTransaction();
+        if (transaction == null) {
+            return false;
+        }
+
+        try {
+            int currentStatus = transaction.getStatus();
+            return currentStatus == Status.STATUS_ACTIVE ||
+                    currentStatus == Status.STATUS_MARKED_ROLLBACK ||
+                    currentStatus == Status.STATUS_PREPARED ||
+                    currentStatus == Status.STATUS_UNKNOWN ||
+                    currentStatus == Status.STATUS_PREPARING ||
+                    currentStatus == Status.STATUS_COMMITTING ||
+                    currentStatus == Status.STATUS_ROLLING_BACK;
+        } catch (SystemException e) {
+            throw new RuntimeException("Error getting the status of the current transaction", e);
+        }
+    }
+
+    private Transaction getCurrentTransaction() {
+        try {
+            return transactionManager.getTransaction();
+        } catch (SystemException e) {
+            throw new RuntimeException("Error getting the current transaction", e);
+        }
+    }
+
+    /**
+     * Representing of the context state. It's a container for all available beans in the context.
+     * It's filled during bean usage and cleared on destroy.
+     */
+    private static class TransactionContextState<T> implements ContextState {
+
+        private final ConcurrentMap<Contextual<T>, ContextInstanceHandle<T>> mapBeanToInstanceHandle = new ConcurrentHashMap<>();
+
+        /**
+         * Put the contextual bean and its handle to the container.
+         *
+         * @param bean bean to be added
+         * @param handle handle for the bean which incorporates the bean, contextual instance and the context
+         */
+        void put(Contextual<T> bean, ContextInstanceHandle<T> handle) {
+            mapBeanToInstanceHandle.put(bean, handle);
+        }
+
+        /**
+         * Remove the bean from the container.
+         *
+         * @param bean contextual bean instance
+         */
+        void remove(Contextual<T> bean) {
+            try {
+                mapBeanToInstanceHandle.remove(bean);
+            } catch (NullPointerException ignore) {
```

Comment:
Ignore `NPE`? If it's meant to be for the maps not supporting null keys, then I don't think your `bean` will ever be null.